app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyResponseDao.java

```diff
@@ -59,25 +49,33 @@ public void setSurveyDao(DynamoSurveyDao surveyDao) {
         this.surveyDao = surveyDao;
     }
     
-    @Override
-    public SurveyResponse createSurveyResponse(GuidCreatedOnVersionHolder keys, String healthCode,
-            List<SurveyAnswer> answers) {
-        return createSurveyResponseInternal(keys, healthCode, answers, BridgeUtils.generateGuid());
+    @Autowired
+    public void setDistributedLockDao(DistributedLockDao lockDao) {
+        this.lockDao = lockDao;
     }
-
+    
     @Override
     public SurveyResponse createSurveyResponse(GuidCreatedOnVersionHolder keys, String healthCode,
             List<SurveyAnswer> answers, String identifier) {
+        
+        String key = RedisKey.SURVEY_RESPONSE_LOCK.getRedisKey(identifier);
+        String lock = null;
```

Comment:
DDB optimistic locking will take care of this for us, so we don't need to use our own locking mechanism.